salt/master.py

```diff
@@ -2268,10 +2347,92 @@ def publish(self, clear_load):
         self._send_ssh_pub(payload, ssh_minions=ssh_minions)
         self._send_pub(payload)
 
-        return {
+        ret = {
             "enc": "clear",
             "load": {"jid": clear_load["jid"], "minions": minions, "missing": missing},
         }
+        if self.opts["order_masters"]:
+            ret["syndics"] = syndics
+        return ret
+
+    def _collect_syndics(self, *, clear_load, extra, minions, auth_list, auth_type):
```

Comment:
That's a good question. I think that there's some mechanism that will time this out higher up the stack. One interesting phenomena I've seen is that the CLI may report that the job didn't really run, even though events actually get published and the minions return back to the syndics which return back to the MoM in a sort of eventual consistency. At the time I *was* timing out on the `while` loop -- which behavior I *could* restore. Though I'm not entirely sure the best timeout to add for that. The `clear_load["to"]` contains a timeout value in seconds, but *other* places up the stack also use that value so... I could use it as well, or use half or some fraction of it? :shrug: My current experience doesn't give me any strong opinions on it.

Reply:
I went ahead and added the timeout logic in there. Should still work the same, and roughly the same as it used to (that is, if the syndic was offline you just kinda get a :shrug: from salt and it may or may not be thrilled about it)